### PR TITLE
Change to use context

### PR DESCRIPTION
**Pull request by AriGute**

### CODE REVIEW

File: src/components/TeamGeneratorPage/ImportPlayerList.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useContext } from 'react';
+import { DisplayContext } from './services/Context';
 
 const ImportPlayerList = ({ callback }) => {
 	const [playerList, setPlayerList] = useState('');
+	const displayContext = useContext(DisplayContext);
 
 	function addPlayerButton() {
 		let list = playerList
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/ImportPlayerList.js:9]: can be const

File: src/components/TeamGeneratorPage/ImportPlayerList.js [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 			.split('\n')
-			.filter((p) => p != '')
+			.filter((p) => p != '' && /[a-zA-Z]/.test(p))
 			.map((p) => p.trim());
 		callback(list);
 	}
+	function clear() {
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/ImportPlayerList.js]: don't use function()
change it to const and give it a much more clear name, maybe onClear

File: src/components/TeamGeneratorPage/TeamGenerator.js [hunk 3]
```diff
@@ -1,52 +1,51 @@
...
-		setPlayerList(TeamsHandler.getAllPlayers());
+		setPlayerList(TeamsHandler.getPublicGroup());
 	}
 
-	function addPlayerToPreTeam(e) {
-		const { player, team } = e.detail;
-		TeamsHandler.addPlayerToPreTeam(player, team);
-		setPlayerList(TeamsHandler.getAllPlayers());
-		window.dispatchEvent(new Event(eventsNames.updateTeams));
+	function loadPlayersFromStorage(params) {
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/TeamGenerator.js]: use const and not function inside a react component

File: src/components/TeamGeneratorPage/TeamGenerator.js [hunk 4]
```diff
@@ -1,52 +1,51 @@
...
-		window.dispatchEvent(new Event(eventsNames.updateTeams));
+	function updateDisplay() {
+		setPlayerList(TeamsHandler.getPublicGroup());
+	}
 
-		return () => {
-			window.removeEventListener(clearListener);
-			window.removeEventListener(addPreTeamListener);
-		};
+	useEffect(() => {
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/TeamGenerator.js:36]: why do you need useEffect? this logic can be handled without useEffect i think

File: src/components/TeamGeneratorPage/TeamGenerator.js [hunk 5]
```diff
@@ -1,52 +1,51 @@
...
+		displayContext.toUpdate.push(updateDisplay);
+		loadPlayersFromStorage();
 	}, []);
 
 	return (
 		<div className='flex flex-col items-center'>
-			<ImportPlayerList callback={importPlayerList}></ImportPlayerList>
-			<TeamCompCard list={playerList} name={'Players'} removeFunc={removePlayer}></TeamCompCard>
-			<TeamsDisplay playerList={playerList}></TeamsDisplay>
+			<DisplayContext.Provider value={displayContext}>
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/TeamGenerator.js:44]: if you wrap your main.js with a provider you don't have to do it here and your whole project will be able to use the useContext

File: src/components/TeamGeneratorPage/TeamGenerator.js [hunk 6]
```diff
@@ -1,52 +1,51 @@
...
+				<ImportPlayerList callback={importPlayerList}></ImportPlayerList>
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/TeamGenerator.js]: don't use callback as name of a prop, give it a real name

File: src/components/TeamGeneratorPage/TeamsDisplay.js [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
-import { eventsNames } from '../../utils/consts';
+import { DisplayContext } from './services/Context';
+import { teams } from '../../utils/consts';
+
 const TeamsDisplay = ({ playerList }) => {
 	const [teamA, setTeamA] = useState([]);
 	const [teamB, setTeamB] = useState([]);
+	const displayContext = useContext(DisplayContext);
 
 	function generateTeamsButton() {
```
Comment (Ronielo) [inline src/components/TeamGeneratorPage/TeamsDisplay.js]: same, change functions to const

**Review by Ronielo — APPROVED** [review]

File: src/components/TeamGeneratorPage/TeamGenerator.js (hunk 6 above)
Comment (AriGute) [reply src/components/TeamGeneratorPage/TeamGenerator.js:45]: it's here on purpose

Comment (Ronielo) [reply src/components/TeamGeneratorPage/TeamGenerator.js]: just change the naming

**Pull request merged**